hw/ip/otbn/doc/_index.md

```diff
@@ -372,14 +372,38 @@ The number of entries in the loop stack is implementation-dependent.
 
 ## Design Details
 
-<div class="bd-callout bd-callout-warning">
-  <h5>Note</h5>
+### Error Handling and Reporting
 
-  To be filled in as we create the implementation.
-</div>
+By design, OTBN is a simple processor and provides no error handling support to code that runs on it.
+Any errors will cause OTBN to stop executing and report the failure to the host processor.
+
+Whenever OTBN observes an error, it will generate an alert.
+This gets sent to the alert manager.
+The alert will either be fatal or recoverable, depending on the class of error: see [Alerts]({{< relref "hw/ip/otbn/doc#alerts" >}}) and [Error Conditions]({{< relref "hw/ip/otbn/doc#error-conditions" >}}) below for details.
+
+If OTBN was running when the alert occurred (this is true whenever {{< regref "STATUS.busy" >}} is high), it will also:
+- Immediately stop fetching and executing instructions.
+- Set {{< regref "INTR_STATE.done" >}} and clear {{< regref "STATUS.busy" >}}, marking the operation as completed.
+- Set the {{< regref "ERR_CODE" >}} register to a non-zero value describing the error.
+
+Note that OTBN can detect some errors even when it isn't running.
+One example of this is an error caused by an ECC failure when reading or programming OTBN's memories over the bus.
+In this case, the {{< regref "ERR_CODE" >}} register will not change.
+This avoids race conditions with the host processor's error handling software.
```

Comment:
You can remove that sentence, since (as the alert section says below) we only trigger fatal alerts if OTBN isn't running (which don't set ERR_CODE anyways).

Instead, add that we only issue fatal alerts in that case.
And I'm not sure if that's the right thing to do, actually. Imagine we have shadow CSRs which require two consecutive writes, do we really want to lock up the whole system if those two writes don't match? Even in development mode?

Reply:
I don't think that's right. For example, we trigger a fatal alert on an ECC failure if OTBN is running. In that case, we stop OTBN and write both the `FATAL_ALERT_CAUSE` register and `ERR_CODE`.

However, if we spot (say) a failed memory read when OTBN isn't running, we mustn't trash `ERR_CODE` (because software might be half-way through dealing with a just-finished OTBN), but will write `FATAL_ALERT_CAUSE`.

Good point about the shadow registers. I guess we'll have to think through that properly when we come to implement them.

Reply:
Isn't this paragraph strictly about the case when OTBN is not running (first sentence of the paragraph)?

Reply:
I discussed this with @rswarbrick and I'm just confused here, ignore my initial comment.

Reply:
:-) Well, it's good that we re-read this. I've updated the text a little, which should hopefuly make it clearer.